File: core/src/main/java/com/fahamutech/kcore/admin/database/connector/IcategoryDataSource.java
package com.fahamutech.kcore.admin.database.connector;

import com.fahamutech.kcore.admin.database.DataBaseCallback;
import com.fahamutech.kcore.admin.model.ICategory;

public interface IcategoryDataSource {
    void newCategory(ICategory category, DataBaseCallback... dataBaseCallbacks);

    void deleteCategory(DataBaseCallback... dataBaseCallbacks);
}

File: core/src/main/java/com/fahamutech/kcore/admin/database/connector/IcategoryDataSource.kt
package com.fahamutech.kcore.admin.database.connector

import com.fahamutech.kcore.admin.database.DataBaseCallback
import com.fahamutech.kcore.admin.model.ICategory

interface IcategoryDataSource {
    fun newCategory(category: ICategory?, vararg dataBaseCallbacks: DataBaseCallback?)
    fun deleteCategory(vararg dataBaseCallbacks: DataBaseCallback?)
}